From 3854a823796605228a5e578630e7c629547dd16f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Wilmet?= <swilmet@mailfence.com>
Date: Fri, 12 Apr 2024 14:38:32 +0200
Subject: [PATCH] PreferencesDialog: remove 'Theme variant' combobox

The setting is still available in the "hidden" GSettings.

The 'Theme variant' is removed from the preferences dialog because the
work is not finished:
- There needs to be two GSettings keys for the style/color scheme.
- When the dark theme variant is selected, it should be impossible to
  choose the classic or tango style schemes (they work only with a light
  theme).
---
 gedit/gedit-preferences-dialog.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/gedit/gedit-preferences-dialog.c b/gedit/gedit-preferences-dialog.c
index 1036ef578..883c6355a 100644
--- a/gedit/gedit-preferences-dialog.c
+++ b/gedit/gedit-preferences-dialog.c
@@ -316,13 +316,10 @@ setup_font_colors_page (GeditPreferencesDialog *dlg)
 {
 	GeditSettings *gedit_settings;
 	GSettings *editor_settings;
-	GSettings *ui_settings;
 	GtkWidget *font_component;
-	GtkWidget *theme_variant_combo_box;
 
 	gedit_settings = _gedit_settings_get_singleton ();
 	editor_settings = _gedit_settings_peek_editor_settings (gedit_settings);
-	ui_settings = _gedit_settings_peek_ui_settings (gedit_settings);
 
 	/* Configure GtkGrid placeholder */
 	gtk_orientable_set_orientation (GTK_ORIENTABLE (dlg->font_and_colors_placeholder),
@@ -336,12 +333,6 @@ setup_font_colors_page (GeditPreferencesDialog *dlg)
 	gtk_container_add (GTK_CONTAINER (dlg->font_and_colors_placeholder),
 			   font_component);
 
-	/* Theme variant */
-	theme_variant_combo_box = tepl_prefs_create_simple_theme_variant_combo_box (ui_settings,
-										    GEDIT_SETTINGS_THEME_VARIANT);
-	gtk_container_add (GTK_CONTAINER (dlg->font_and_colors_placeholder),
-			   theme_variant_combo_box);
-
 	/* Color/Style scheme */
 	setup_font_colors_page_style_scheme_section (dlg);
 }
-- 
GitLab

